src/compat/compat.h

```diff
@@ -711,3 +711,14 @@ initReadOnlyStringInfo(StringInfo str, char *data, int len)
 	str->cursor = 0;
 }
 #endif
+
+/* PG18 introduces new CompareType for ordering operations
+ * Add macros so we can use the new naming for older versions.
+ * https://github.com/postgres/postgres/commit/8123e91f
+ */
+#if PG18_LT
+#define CompareType int16
+#define COMPARE_LT BTLessStrategyNumber
+#define COMPARE_GT BTGreaterStrategyNumber
+#define pk_cmptype pk_strategy
```

Comment:
Let's make this upper case too, I can imagine unexpected effects like creating a local variable named like this and then not being able to find it in the debugger.